### ISSUE TITLE
Characters are not marked as offline in characters.online

### ISSUE THREAD

**Issue by d-christ** [azerothcore/azerothcore-wotlk#20545]
### Current Behaviour

When a character loggs in and then logs out it is still marked as online in the "characters" table (online is on 1). Its only reset when the server is restarted.

### Expected Blizzlike Behaviour

Characters who log off should be offline.

### Steps to reproduce the problem

1. Log in
2. Log out
3. Check characters table for that character, value vor "online" is 1, not 0.

### AC rev. hash/commit

AC> AzerothCore rev. unknown 1970-01-01 00:00:00 +0000 (Archived branch) (Unix, Release, Static)
Connected players: 0. Characters in world: 1369.
Connection peak: 1.
Server uptime: 12 hour(s) 47 minute(s) 45 second(s)
Update time diff: 10ms. Last 500 diffs summary:
|- Mean: 13ms
|- Median: 11ms
|- Percentiles (95, 99, max): 26ms, 30ms, 32ms
Using SSL version: OpenSSL 3.0.2 15 Mar 2022 (library: OpenSSL 3.0.2 15 Mar 2022)
Using Boost version: 1.74.0
Using CMake version: 3.22.1
Using MySQL version: 80040
Found MySQL Executable: /usr/bin/mysql
Compiled on: Linux 5.15.0-125-generic
Worldserver listening connections on port %u
Realmlist (Realm Id: 1) configured in port 8085
VMAPs status: Enabled. LineOfSight: true, getHeight: true, indoorCheck: true
MMAPs status: Enabled
maps directory located in /home/azerothcore/azerothcore-wotlk/data/maps. Total size: 291014951 bytes
vmaps directory located in /home/azerothcore/azerothcore-wotlk/data/vmaps. Total size: 658130721 bytes
mmaps directory located in /home/azerothcore/azerothcore-wotlk/data/mmaps. Total size: 2192910844 bytes
Default DBC locale: enUS.
All available DBC locales: enUS
Using World DB: ACDB 335.12-dev
Using Playerbots DB Revision:
Latest LoginDatabase update: 2024_01_20_00.sql
Latest CharacterDatabase update: playerbots_names.sql
Latest WorldDatabase update: world_playerbots_rpg_races.sql
LoginDatabase queue size: 0
CharacterDatabase queue size: 0
WorldDatabase queue size: 0
PlayerbotsDatabase queue size: 0
List of enabled modules:
|- mod-ah-bot
|- mod-aoe-loot
|- mod-breaking-news-override
|- mod-eluna
|- mod-guildhouse
|- mod-playerbots
|- mod-transmog

### Operating system

Ubuntu 22

### Custom changes or Modules

Eluna (not used, just installed, AH-Bot, AOE-loot (no changes), guildhouse, playerbots (1000 Bots), transmog, breaking news.

Except for the more playerbots all plugins are just installed and are used "as is".

**Comment by TheSCREWEDSoftware** [comment]
Can't replicate issue this issue, fix your harsh and report this issue to PlayerBots github.

**Comment by noisiver** [comment]
This is **not** specific to playerbots. It happens without it, it always has. If you look through the code you will find that characters are **only** marked as offline (online = 1) on server startup. NEVER when they actually log out. There is a reason why there's a draft PR addressing this issue here.

**Comment by noisiver** [comment]
I found one instance where it fails to update the online status.

With a GM account it's fine in any situation I tested because they have instant logout.
With a regular account, waiting for the logout timer or being in an area where you instantly log out it seems to reset to 0 fine too.
With a regular account, selecting `Exit Game` and then `Exit now` will prevent it from updating the online status properly. This can't be tested in any area where you instantly log out.
